File: perfect_Number.py

#PROGRAMMING QUESTION - Moses VB

# 1. PERFECT NUMBERS
# This python code returns whether a given number N is a perfect number or not

#Function to do the calculation
def perfect_number(number):
    total_n = 0
    for i in range(1, number):
        if(number % i == 0):
            total_n = total_n + i
    return total_n        
 
# Taking input from the user
number = int(input("Please Enter any number to check: "))

#calling the defined function above using if statement
if (number == perfect_number(number)):
    print("\n %d is a Perfect Number entered" %number)
else:
    print("\n %d is not a Perfect Number, try another number" %number)
	
#####

File: java/perfectNumber.java
//PROGRAMMING QUESTION - Moses VB

//1. PERFECT NUMBERS
// This Java code returns whether a given number N is a perfect number or not

import java.util.Scanner;
 
public class PerfectNumber {
 
    // The main method which begins the execution of this class
    public static void main(String args[]) {
 
		//declaration of variables
        int inputValue;
		int totalN = 0;
 
		//using scanner class to accept user input
        Scanner userInput = new Scanner(System.in);
        System.out.print("Enter a number greater than to check: ");
		
        // Take entered value and store it into the  inputValue variable
        inputValue = userInput.nextInt();
       
        // check/compare the input number whether it's a perfect number of not
        for(int i = 1; i <= inputValue; i++) {
            if(inputValue%i==0) 
			totalN += i;  //or totalN = totalN+i;
        }
		
		//check and display message to the user
        if(totalN == inputValue)
            System.out.print("The number "+inputValue+" is a perfect number ");
        else 
           System.out.print("The number "+inputValue+" is not a perfect number, try again "); 
        
    } 
}

